x-pack/solutions/security/plugins/elastic_assistant/server/ai_assistant_service/index.ts

```diff
@@ -229,29 +245,124 @@ export class AIAssistantService {
         pluginStop$: this.options.pluginStop$,
       });
 
-      await this.knowledgeBaseDataStream.install({
-        esClient,
-        logger: this.options.logger,
-        pluginStop$: this.options.pluginStop$,
-      });
+      if (this.assistantDefaultInferenceEndpoint) {
+        const knowledgeBaseDataStreamExists = await esClient.indices.getDataStream({
+          // @ts-expect-error
+          name: this.knowledgeBaseDataStream.name,
```

Comment:
Let's make this guy `public readonly name: string` instead of adding all these  // @ts-expect-error

https://github.com/elastic/kibana/blob/32e9a990a9b8633ea212e4c36b72edcc47ea2a0c/x-pack/solutions/security/packages/index-adapter/src/index_adapter.ts#L55

Reply:
or call
```suggestion
          name: this.knowledgeBaseDataStream..getIndexName(`*`),
```